app/src/common/hooks/useFhirResourceTreeData.ts

```diff
@@ -217,18 +232,46 @@ export const buildTree = (
     );
   }
 
-  if (
-    isElementNodeChildOf(currentElementNode, previousElementNode) &&
-    (!previousElementNode.isArray || rootNode === previousElementNode)
-  ) {
-    previousElementNode.children.push(currentElementNode);
-    buildTree(rest, rootNode, currentElementNode, parentPath);
+  if (isElementNodeChildOf(currentElementNode, previousElementNode)) {
+    if (
+      previousElementNode.isArray &&
+      previousElementNode.type === "BackboneElement"
+    ) {
+      buildTree(rest, rootNode, previousElementNode, parentPath);
+    } else {
+      previousElementNode.children.push(currentElementNode);
+      buildTree(rest, rootNode, currentElementNode, parentPath);
+    }
   } else {
     const parent = getParent(previousElementNode, rootNode);
     if (parent) buildTree(elementDefinitions, rootNode, parent, parentPath);
   }
 };
 
+/**
+ * Add Attributes to RootNode array elements
+ * @param rootNode Root of the ElementNode tree
+ * @param attributeNodes attributes to add to RootNode
+ */
+const addAttributesToTree = (
+  rootNode: ElementNode,
+  attributeNodes: ElementNode[]
+) => {
+  attributeNodes.forEach((attribute) => {
+    if (attribute.definition.path) {
+      const node = getNode(
+        "path",
+        attribute.path.split(/[[]\d+]$/).join(""),
+        rootNode
+      );
+      if (node) {
+        node.children.push(attribute);
+        node.children.sort((a, b) => (a.path > b.path ? 1 : -1));
```

Comment:
Maybe it would be better to sort the `attributeNodes` array before the `forEach`, once and for all

Reply:
Actually, we don't need this sorting function as attributes will be added in the right order when clicking on the add icon.

Reply:
Can't be sure you'll get them in the right order after the fetch